app/scripts/modules/core/src/application/application.model.ts

```diff
@@ -126,10 +126,12 @@ export class Application {
 
     const noneLeftToFetch = statuses.every(({ status }) => ['FETCHED', 'NOT_INITIALIZED'].includes(status));
     const lastFetch = statuses.reduce((latest, status) => Math.max(latest, status.lastRefresh || 0), 0);
+    const allLoaded = statuses.every(({ loaded }) => loaded);
 
     return {
       status: rolledUpStatus,
       lastRefresh: noneLeftToFetch ? lastFetch : this.lastRefresh,
+      loaded: allLoaded,
```

Comment:
is this related/required for the other data source changes? I don't fully understand its purpose right now

Reply:
This is just because we added `loaded` to `IFetchStatus` and this method was using `IFetchStatus` as its return type. Seemed better and simple enough to just give a loaded rollup instead of just doing `loaded: undefined` like what it's already doing for `data`.

Reply:
AHA ok. i get it now